feat(servicenow): add a Servicenow tickets tab on entity page

ciiay opened the pull request:
## Hey, I just made a Pull Request!

<!-- Please describe what you added, and add a screenshot if possible.
     That makes it easier to understand the change so we can :shipit: faster. -->

#### :heavy_check_mark: Checklist

<!--- Please include the following in your Pull Request when applicable: -->

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/community-plugins/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/community-plugins/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

Screen recording:


https://github.com/user-attachments/assets/dd88514b-b187-4568-b63e-dcd85a352cde

awanlin reviewed (dismissed):
Hi @ciiay, would it be possible to refactor this to not include a full backstage app and follow the `dev/index` pattern? We really much prefer that 👍

BethGriggs commented on workspaces/servicenow/package.json:
```diff
@@ -0,0 +1,68 @@
+    "lint": "backstage-cli repo lint --since origin/main",
+    "lint:all": "backstage-cli repo lint",
+    "prettier:check": "prettier --check .",
+    "new": "backstage-cli new --scope @backstage-community"
```
```suggestion
    "new": "backstage-cli new --scope @backstage-community",
    "postinstall": "cd ../../ && yarn install"
```

its-mitesh-kumar reviewed (commented):
@ciiay is screen recording in the PR description is for this PR ?

its-mitesh-kumar reviewed (commented):
Tested in the Dev mode
Here I see two issue but when I was testing in real environment it was not there. 
1. For the first time changing page size from 5-> 10 not working. For 2nd time and all it's working.

https://github.com/user-attachments/assets/e6bbf7bf-c678-41e4-8134-65175183576b

2. Action button not working
 

https://github.com/user-attachments/assets/76adf608-28cc-4eb6-947e-d1b76be9181b


Other functionality looks good to me .

https://github.com/user-attachments/assets/22cf4617-2b22-4fc5-891c-f6e9f4c8d10a

ciiay commented:
Hi @its-mitesh-kumar , thanks for the review. Updated the screenshot video and fixed the UI issue.

its-mitesh-kumar reviewed (approved):
Tested in dev mode seems working as expected.

https://github.com/user-attachments/assets/2d930414-b872-4253-8bea-e76ba0406636

christoph-jerolimov reviewed (approved):
Thanks @ciiay , ~~still testing~~ tested some cases, works fine! lgtm! 👍

@Parsifal-M can you take a look since Andre is on vacation. :palm_tree:

Fyi, this is the first version of this ServiceNow plugin, but we will continue to develop and maintain the plugin, there is more to come. :smile:

Parsifal-M commented:
I should have some time to give this a look over tomorrow! 🙏

Parsifal-M commented on workspaces/servicenow/plugins/servicenow/package.json:
```diff
@@ -0,0 +1,88 @@
+  "files": [
+    "dist"
+  ],
+  "scalprum": {
+    "name": "backstage-community.plugin-servicenow",
+    "exposedModules": {
+      "PluginRoot": "./src/index.ts",
+      "isMyProfile": "./src/utils/isMyProfile.ts"
+    }
+  },
```
What is this? Haven't seen this before 😅 does it need to be here?

christoph-jerolimov replied:
I believe you can export the isMyProfile function also in `workspaces/servicenow/plugins/servicenow/src/index.ts`

```tsx
export { servicenowPlugin, ServicenowPage } from './plugin'; // I personally like to export everything from ./plugin
export { isMyProfile } from './utils/isMyProfile';
```

ciiay replied:
This was something I tried which didn't work. I already found the correct way to use the condition module in our dynamic plugin system. I have had this removed 👍

Parsifal-M reviewed (commented):
Heya! Had a bit of a look and prod around, did not in all honesty dive too deeply into everything because I don't really know what servicenow looks like, but I checked a lot of the best practises for building Backstage plugins and it looks good 😄 

Just a couple of comments that I don't think are blocking but would like to see what you think! 👍

ciiay commented:
Hi @Parsifal-M , thanks for taking time on my PR. I have updated all the README.md files and updated the condition module. Please take a look when you have time. Thanks in advance 🙌
